Fix python bindings ESmry, start date

tskille opened the pull request:
This PR fixes handling of daylight-saving time when importing start date from a summary file (and ESMRY) using python bindings.

Consider the following example. Start date is set in the data deck. 

```
-- daylight saving from 2020-03-29 02:00 -> 03:00
START
 29 'MAR' 2020 05:04:41 /
```

When running this deck with OPM Flow, the following is saved to the SMSPEC file

```
'STARTDAT'           6 'INTE'
          29           3        2020           5           4    41000000
```

So far, all good. 

However, when using the following python script to import start date from the summary file

```python
import sys

from opm.io.ecl import ESmry

if __name__ == "__main__":

    smry = ESmry(sys.argv[1])

    print ("\n start date > {} \n".format(smry.start_date))
```

results are as following

```
start date > 2020-03-29 06:04:41
```

This is not as expected. As could be seen, this is one hour shifted relative to the input and to what is stored in the SMSPEC file.

lisajulia reviewed (approved):
@tskille: Thanks for spotting and fixing this! :+1: 
Can you possibly add the test from the description of the PR as an actual test to the opm tests?

tskille commented:
Thanks @lisajulia. I have now added one extra regression test in test_esmry.py.  This is testing both .SMSPEC and .ESMRY files.  

I also found one new error in opm/io/eclipse/ExtESmry.cpp which I fixed. 

Hopefully this is ready to go in now.

lisajulia commented:
@tskille: Thanks!

lisajulia commented:
jenkins build this please
